### DISCUSSION TITLE
How many times can I software restart ESP32 ?

### DISCUSSION THREAD

**Discussion by dharmikP17** [espressif/arduino-esp32#8223] [Q&A] [answered]
Is there any limit or recommended times I should use software restart in esp32 ?
I have application where I am trying to implement software restart in ESP32 in like every 15 seconds. If I plan to do this for years, will there be any problem ?
Will it affect flash memory of ESP32 ? Is restart caused by power removal is different from software restart ?

**Comment by mrengineer7777** [comment, marked as answer]
Short answer: there is no limit.

Discussion:
15 seconds is pretty short.  This would not play well with WiFi or Ethernet connections.  Are you using the module offline?
Flash memory should not be affected unless you are saving to NVS or writing a file before every restart.

The module seems to have identical startup for power and software resets.  With a software reset it's possible to set certain IRAM(?) variables to persist state across reboots.  That's the mechanism IDF uses to persist crash info.  Example https://github.com/espressif/arduino-esp32/pull/8105.

I'm curious.  Why restart every 15s?

**Reply by dharmikP17** [reply]
Thank you for answering.

I am using Wi-Fi. If I make sure to turn off Wi-Fi radio module still it will have problem ?
I am not writing new file at restart so I guess there will be only reading from memory so that will not be an issue I suppose.

I do not need to check restart status as of now. But in time I guess I will have to, so thank you for sharing that too.

Reason is that I want to use Wi-Fi and Bluetooth both hence need to turn off both modules again and again. It turns out that bluetooth cant be restarted once turned off specifically BLE and same for A2DP. I have opened issue for that as well but did not get any response in a month so I am trying to do alternate approach with reboot.

**Reply by mrengineer7777** [reply]
I take a similar approach when switching from Ethernet to WiFi.

If you stop wifi before rebooting then you should be ok. I used to have issue with my router getting mad at my device (failing to reconnect) if I rebooted the ESP32 while connected. I use something like wifi.mode(none) in my reboot routine.

I'm familiar with the current Bluetooth issues. Perhaps if I have free time (ha!) in a couple months I'll dig into the libraries and see what I can fix.

**Reply by dharmikP17** [reply]
You can refer to issue I have posted. https://github.com/espressif/arduino-esp32/issues/7990
Thank you.
